Do not translate file extensions

hebasto opened the pull request:
File extensions are untranslatable by their nature.

jarolrod reviewed (commented):
ACK 6af0a9a9e59b17e6b8491bc9715b67b5310e599e

Nice catch, this will help prevent any unintended translations

jonatack reviewed (commented):
Tested ACK 6af0a9a9e59b17e6b8491bc9715b67b5310e599e

Two questions: (a) does this change require translators to update every language? (b) Testing this patch, I don't see the file extensions in the dialogs. I then built master and I don't see the extensions there either :thinking: 

```
src/qt/locale/bitcoin_fr.ts:1574:        <source>Partially Signed Transaction (Binary) (*.psbt)</source>
src/qt/locale/bitcoin_fr.ts:1575:        <translation>Transaction signée partiellement (fichier binaire) (*.psbt)</translation>
src/qt/locale/bitcoin_fr.ts:2557:        <source>Partially Signed Transaction (Binary) (*.psbt)</source>
src/qt/locale/bitcoin_fr.ts:2558:        <translation>Transaction signée partiellement (fichier binaire) (*.psbt)</translation>
src/qt/locale/bitcoin_fr.ts:3496:        <source>Partially Signed Transaction (*.psbt)</source>
src/qt/locale/bitcoin_fr.ts:3497:        <translation>Transaction signée partiellement (*.psbt)</translation>
```

hebasto commented:
> (a) does this change require translators to update every language?

Yes.

> (b) Testing this patch, I don't see the file extensions in the dialogs. I then built master and I don't see the extensions there either thinking (checked in default and in French)

See `GUIUtil::getSaveFileName` implementation.

luke-jr commented on src/qt/addressbookpage.cpp:
```diff
@@ -293,9 +293,11 @@ void AddressBookPage::done(int retval)
         tr("Export Address List"), QString(),
-        tr("Comma separated file (*.csv)"), nullptr);
+        // TODO: Replace QStringLiteral with QLatin1String when Qt minimum version >= 5.10.
+        tr("Comma separated file (%1)").arg(QStringLiteral("*.csv")), nullptr);
```
IIRC the parenthesis format is also special, to trigger the dialog filtering files shown. So perhaps this should be:
```suggestion
        tr("Comma separated file") + QStringLiteral(" (*.csv)"), nullptr);
```

hebasto replied:
Thank you! [Updated](https://github.com/bitcoin-core/gui/pull/220#issuecomment-785364090).

hebasto commented:
Updated 6af0a9a9e59b17e6b8491bc9715b67b5310e599e -> 5fe0ff342b76c4557450a43f27bdd99e48a3e966 ([pr220.01](https://github.com/hebasto/gui/commits/pr220.01) -> [pr220.02](https://github.com/hebasto/gui/commits/pr220.02), [diff](https://github.com/hebasto/gui/compare/pr220.01..pr220.02)):

- addressed @luke-jr's [comment](https://github.com/bitcoin-core/gui/pull/220#discussion_r582144001)
- minimized diff

luke-jr commented on src/qt/psbtoperationsdialog.cpp:
```diff
@@ -141,7 +141,7 @@ void PSBTOperationsDialog::saveTransaction() {
     QString filename = GUIUtil::getSaveFileName(this,
         tr("Save Transaction Data"), filename_suggestion,
-        tr("Partially Signed Transaction (Binary) (*.psbt)"), &selected_filter);
+        tr("Partially Signed Transaction (Binary)") + QLatin1String(" (*.psbt)"), &selected_filter);
```
Annotate the `tr` here and later too?

luke-jr replied:
```suggestion
        tr("Partially Signed Transaction (Binary)", "Name of binary PSBT file format") + QLatin1String(" (*.psbt)"), &selected_filter);
```

hebasto replied:
Thanks! [Done](https://github.com/bitcoin-core/gui/pull/220#issuecomment-785840104).

hebasto commented:
Could you suggest a good variant of annotation?

On Thu, 25 Feb 2021 at 00:18 Luke Dashjr <notifications@github.com> wrote:

> *@luke-jr* commented on this pull request.
> ------------------------------
>
> In src/qt/psbtoperationsdialog.cpp
> <https://github.com/bitcoin-core/gui/pull/220#discussion_r582345696>:
>
> > @@ -141,7 +141,7 @@ void PSBTOperationsDialog::saveTransaction() {
>      filename_suggestion.append(".psbt");
>      QString filename = GUIUtil::getSaveFileName(this,
>          tr("Save Transaction Data"), filename_suggestion,
> -        tr("Partially Signed Transaction (Binary) (*.psbt)"), &selected_filter);
> +        tr("Partially Signed Transaction (Binary)") + QLatin1String(" (*.psbt)"), &selected_filter);
>
> Annotate the tr here and later too?
>
> —
> You are receiving this because you authored the thread.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin-core/gui/pull/220#pullrequestreview-597986728>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH3PXPWTEF7VBYA6M57P7L3TAV3MDANCNFSM4YCWAUUA>
> .
>
-- 

--
Hennadii Stepanov

hebasto commented:
Updated 5fe0ff342b76c4557450a43f27bdd99e48a3e966 -> d4ed848e00cc0232cae234f0361b54bb7375fe2f ([pr220.02](https://github.com/hebasto/gui/commits/pr220.02) -> [pr220.03](https://github.com/hebasto/gui/commits/pr220.03), [diff](https://github.com/hebasto/gui/compare/pr220.02..pr220.03)):

- added disambiguation strings for PSBT translations as [suggested](https://github.com/bitcoin-core/gui/pull/220#discussion_r582375795) by @luke-jr

jonatack commented:
re-ACK d4ed848e00cc0232cae234f0361b54bb7375fe2f

(IDK if the wallet  `*.dat` ts strings should be annotated as well or if it's self-evident.)

luke-jr commented on src/qt/walletview.cpp:
```diff
@@ -273,7 +273,7 @@ void WalletView::backupWallet()
     QString filename = GUIUtil::getSaveFileName(this,
         tr("Backup Wallet"), QString(),
-        tr("Wallet Data (*.dat)"), nullptr);
+        tr("Wallet Data") + QLatin1String(" (*.dat)"), nullptr);
```
Also needs annotation...

(If you were asked to translate "Wallet Data" with no context, would you know what it was referring to? XD)

hebasto replied:
[Updated](https://github.com/bitcoin-core/gui/pull/220#issuecomment-792023727).

Talkless commented:
Concept ACK. I agree to @luke-jr that one annotation is missing.

hebasto commented:
Updated d4ed848e00cc0232cae234f0361b54bb7375fe2f -> 88df300f20da02060694cfe643e1c882efaa306c ([pr220.03](https://github.com/hebasto/gui/commits/pr220.03) -> [pr220.04](https://github.com/hebasto/gui/commits/pr220.04), [diff](https://github.com/hebasto/gui/compare/pr220.03..pr220.04)).

Now each of changed translation has own disambiguation string.

Talkless reviewed (approved):
tACK 88df300f20da02060694cfe643e1c882efaa306c, tested on Debian Sid with Qt 5.15.2. Tested all filters except for .psbt.

jarolrod reviewed (commented):
re-ACK 88df300f20da02060694cfe643e1c882efaa306c

when to use and how to place disambiguation strings is another thing to add to the Qt dev notes

hebasto commented:
@luke-jr @laanwj 

Some things appear unclear to me.

1) Does the non-default `disambiguation` argument, which is passed to the [`tr`](https://doc.qt.io/qt-5/qobject.html#tr) function, is visible to translators on [transifex](https://www.transifex.com/bitcoin/bitcoin)?

2) What are other options to make designated notes to translators on [transifex](https://www.transifex.com/bitcoin/bitcoin)?
Do ["Translator Comments"](https://doc.qt.io/qt-5.12/i18n-source-translation.html#translator-comments) work in such a manner?

---

In the meantime, converting this pr to a draft.

laanwj commented:
Concept and code review ACK 88df300f20da02060694cfe643e1c882efaa306c

> Does the non-default disambiguation argument, which is passed to the tr function, is visible to translators on transifex?
> What are other options to make designated notes to translators on transifex?
Do "Translator Comments" work in such a manner?

I think those are great questions. I do not  know the answer though. Maybe @wtogami knows who can weigh in—from what i remember he knows some people at Transifex.

hebasto commented:
Even if disambiguation arguments won't actually help translators, they won't hurt.

As already many ACKs are collected, reverted back to "Ready for review".

laanwj commented:
At least the comment makes it into the xml, that's one part of it being picked up
```patch
     <message>
         <location line="+1"/>
-        <source>Comma separated file (*.csv)</source>
+        <source>Comma separated file</source>
+        <comment>Name of CSV file format</comment>
         <translation type="unfinished"></translation>
     </message>
```
